http3/client.go

```diff
@@ -367,3 +378,10 @@ func (c *ClientConn) doRequest(req *http.Request, str *RequestStream) (*http.Res
 	res.Request = req
 	return res, nil
 }
+
+// Conn returns the underlying HTTP/3 connection.
+// This method is only useful for advanced use cases, such as when the application needs to
+// open streams on the HTTP/3 connection (e.g. WebTransport).
+func (c *ClientConn) Conn() *Conn {
+	return c.conn
+}
```

Comment:
I'm not a big fan of this API, but it's (currently) needed by WebTransport.

When we overhaul the stream hijacking API (#4405), we should revisit if this is still necessary, or if there are any better options.